Compfiles/Imo1967P3.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2025 Roozbeh Yousefzadeh. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Roozbeh Yousefzadeh
+-/
+
+import Mathlib.Data.Nat.Prime.Factorial
+import Mathlib.Tactic
+
+import ProblemExtraction
+
+problem_file {
+  tags := [.NumberTheory]
+  importedFrom := "https://github.com/roozbeh-yz/IMO-Steps/blob/main/Lean_v20/imo_proofs/imo_1967_p3.lean"
+}
+
+/-!
+# International Mathematical Olympiad 1967, Problem 3
+
+Let $k, m, n$ be natural numbers such that m + k + 1 is a prime greater than n + 1. Let c(s) = s * (s+1).
+Prove that the product (c(m+1) - c(k)) * (c(m+2) - c(k)) * ... * (c(m+n) - c(k)) is divisible by the product c(1) * c(2) * ... * c(n).
+-/
+
+namespace Imo1967P3
+
+
+problem imo_1967_p3
+  (k m n : ℕ)
+  (c : ℕ → ℕ)
+  (h₀ : 0 < k ∧ 0 < m ∧ 0 < n)
+  (h₁ : ∀ s, c s = s * (s + 1))
+  (h₂ : Nat.Prime (k + m + 1))
+  (h₃ : n + 1 < k + m + 1) :
+  (∏ i ∈ Finset.Icc 1 n, c i) ∣ (∏ i ∈ Finset.Icc 1 n, (c (m + i) - c k)) := by
```

Comment:
Do we know that `c (m + i)` is always greater than or equal to `c k`? If not, then I think we need to cast to the integers do to `c (m + i) - c k` correctly.